core/domain/exp_services.py

```diff
@@ -360,206 +363,383 @@ def apply_change_list(exploration_id, change_list):
-                        for hint_dict in change.new_value
-                    ]
-                    state.update_interaction_hints(new_hints_list)
-                elif (change.property_name ==
-                      exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION):
-                    new_solution = None
-                    if change.new_value is not None:
-                        new_solution = state_domain.Solution.from_dict(
-                            state.interaction.id, change.new_value)
-                    state.update_interaction_solution(new_solution)
-                elif (change.property_name ==
-                      exp_domain.STATE_PROPERTY_SOLICIT_ANSWER_DETAILS):
-                    if not isinstance(change.new_value, bool):
-                        raise Exception(
-                            'Expected solicit_answer_details to be a ' +
-                            'bool, received %s' % change.new_value)
-                    state.update_solicit_answer_details(change.new_value)
-                elif (change.property_name ==
-                      exp_domain.STATE_PROPERTY_CARD_IS_CHECKPOINT):
-                    if not isinstance(change.new_value, bool):
-                        raise Exception(
-                            'Expected card_is_checkpoint to be a ' +
-                            'bool, received %s' % change.new_value)
-                    state.update_card_is_checkpoint(change.new_value)
-                elif (change.property_name ==
-                      exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS):
-                    if not isinstance(change.new_value, dict):
-                        raise Exception(
-                            'Expected recorded_voiceovers to be a dict, '
-                            'received %s' % change.new_value)
-                    # Explicitly convert the duration_secs value from
-                    # int to float. Reason for this is the data from
-                    # the frontend will be able to match the backend
-                    # state model for Voiceover properly. Also js
-                    # treats any number that can be float and int as
-                    # int (no explicit types). For example,
-                    # 10.000 is not 10.000 it is 10.
-                    new_voiceovers_mapping = (
-                        change.new_value['voiceovers_mapping'])
-                    language_codes_to_audio_metadata = (
-                        new_voiceovers_mapping.values())
-                    for language_codes in language_codes_to_audio_metadata:
-                        for audio_metadata in language_codes.values():
-                            audio_metadata['duration_secs'] = (
-                                float(audio_metadata['duration_secs'])
+
+    backend_version = exp_fetchers.get_exploration_by_id(exploration_id).version
+    if (backend_version == frontend_version or
+            frontend_version is None):
```

Comment:
Can you add some explanation above this if condition please e.g. why do we need it and how is it different from the else part? Should both blocks be kept "in sync" when a field is added / modified?

Reply:
In what scenario will frontend_version be None?

Reply:
Done.

Reply:
Frontend_version will be none sometimes when the user is working on the latest version.